### PR TITLE
8255860: Clean up CDS logging related to lambda

### PR DESCRIPTION
**Pull request by iklam**
A small clean up to make the logging more manageable.

[1] Moved some debug printing from `tty->print_cr()` to `-Xlog:cds+lambda=debug`
[2] Moved some logs from `-Xlog:cds` to `-Xlog:cds+lambda`

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8255860](https://bugs.openjdk.java.net/browse/JDK-8255860): Clean up CDS logging related to lambda

### Reviewers
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**)
 * [Yumin Qi](https://openjdk.java.net/census#minqi) (@yminqi - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1062/head:pull/1062`
`$ git checkout pull/1062`

### CODE REVIEW

File: src/hotspot/share/classfile/classListParser.cpp [hunk 1]
```diff
@@ -461,7 +461,6 @@ bool ClassListParser::is_matching_cp_entry(constantPoolHandle &pool, int cp_inde
 }
 
 void ClassListParser::resolve_indy(Symbol* class_name_symbol, TRAPS) {
-
```
Comment (calvinccheung) [inline src/hotspot/share/classfile/classListParser.cpp:464]: Line deleted by accident?

File: test/hotspot/jtreg/runtime/cds/appcds/BadBSM.java [hunk 2]
```diff
@@ -42,7 +42,8 @@ public static void main(String[] args) throws Exception {
 
     OutputAnalyzer out = TestCommon.dump(appJar,
         TestCommon.list("WrongBSM",
-                        "@lambda-proxy WrongBSM 7"));
+                        "@lambda-proxy WrongBSM 7"),
+        "-Xlog:cds+lambda=debug");
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/BadBSM.java:46]: Please align line 46 with 45.

**Review by calvinccheung — APPROVED** [review]
The cleanup looks good. Just couple of nits below.

File: src/hotspot/share/classfile/classListParser.cpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/classfile/classListParser.cpp:464]: It's uncommon to have a blank line at the beginning of a function, so I deleted it.

File: test/hotspot/jtreg/runtime/cds/appcds/BadBSM.java (hunk 2 above)
Comment (iklam) [reply test/hotspot/jtreg/runtime/cds/appcds/BadBSM.java:46]: The `"-Xlog:cds+lambda=debug"` parameter is actually at the same level as the `TestCommon.list("WrongBSM",...)` parameter. Aligning it with line 45 would make it look like an argument for the `TestCommon.list()` call.

**Review by yminqi — APPROVED** [review]
LGTM

**Comment by iklam** [comment]
Thanks @calvinccheung and @yminqi for the review!
/integrate

**Pull request closed without merging**